### ISSUE TITLE
Simplify CellTemplate in XAML

### ISSUE THREAD

**Issue by Symbai** [dotnet/wpf#2147]
Would it be possible to automatically assume the content of CellTemplate is a DataTemplate? I think this is what is the case for 95%. Honestly I haven't seen any other case yet.

I.e. instead of this:
```
<DataGridTemplateColumn.CellTemplate>
    <DataTemplate>
        <TextBlock />
    </DataTemplate>
</DataGridTemplateColumn.CellTemplate>
```

You can write:
```
<DataGridTemplateColumn.CellTemplate>
    <TextBlock />
</DataGridTemplateColumn.CellTemplate>
```

Unless specified otherwise.

**Comment by weltkante** [comment]
I don't think the XAML parser allows leaving out `DataTemplate`. Leaving it out would mean you are assigning `TextBlock` (or rather, arbitrary `UIElements`) to the `CellTemplate` property directly.

Also you are going to confuse the designer even more than it is already, since DataTemplate is an indicator for a context switch (actually you often place a `DataType` attribute on it to help the designer figure out its intellisense). Removing DataTemplate and making it an implicitely context swtich in the CellTemplate property is going to be really hard to support in design time intellisense (when you want to put bindings on your UIElements the designer won't have any clue you've switched context and it needs to show different intellisense)

**Comment by Symbai** [comment]
> 
> 
> I don't think the XAML parser allows leaving out `DataTemplate`. Leaving it out would mean you are assigning `TextBlock` (or rather, arbitrary `UIElements`) to the `CellTemplate` property directly.

I think you misunderstood my request, or I am misunderstanding your answer.

Anyway like I said, the parser / designer /whatssover can automatically add a datatemplate when its missing. I'm not going to remove the datatemplate. I am going to remove the need of developers having to add/type these extra lines in xaml for 95% of all cases and let WPF do this job instead.

Similar as #84

**Comment by weltkante** [comment]
Nope, not similar at all. #84 fits into the existing parser/designer design, it is just marking the property as default. This doesn't work here for the reasons I described above, #84 would correspond to removing `DataGridTemplateColumn.CellTemplate` tag not the `DataTemplate` tag.

I'm not saying its impossible to implement your suggestion but it would require changing how parser/designer works currently.
